uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,453 @@ define([
+        vm.refresh(++vm.pagination.page);
+      }
+    };
+
+    /**
+     * Returns an array of a given size
+     *
+     * @param {number} n - no of elements in the array
+     * @return {Array}
+     */
+    vm.range = function (n) {
+      return new Array(n || 0);
+    };
+
+    /**
+     * Refreshes the leave request data
+     *
+     * @param {string} page - page number of the pagination element
+     */
+    vm.refresh = function (page) {
+      page = page ? page : 1;
+      loadAllRequests(page);
+    };
+
+    /**
+     * Refreshes the leave request data and also changes current selected leave status
+     *
+     * @param {string} status - status to be selected
+     */
+    vm.refreshWithFilter = function (status) {
+      vm.filters.leaveRequest.leaveStatus = status;
+      vm.refresh();
+    };
+
+    /**
+     * Calculates the total number of pages for the pagination
+     *
+     * @return {number}
+     */
+    vm.totalNoOfPages = function () {
+      return Math.ceil(vm.leaveRequests.table.total / vm.pagination.size);
+    };
+
+    (function init() {
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadRegions(),
+        loadDepartments(),
+        loadLocations(),
+        loadLevelTypes(),
+        loadStatuses()
+      ])
+        .then(function () {
+          vm.loading.page = false;
+          loadAllRequests(1);
+        })
+    })();
+
+    /**
+     * Returns the filter object for contacts api
+     *
+     * @return {Object}
+     */
+    function contactFilters() {
+      var filters = vm.filters.contact;
+
+      return {
+        region: filters.region ? filters.region.value : null,
+        department: filters.department ? filters.department.value : null,
+        level_type: filters.level_type.length ? {
+            "IN": filters.level_type.map(function (data) {
+              return data.value;
+            })
+          } : null,
+        location: filters.location ? filters.location.value : null
+      };
+    }
+
+    /**
+     * Loads the absence periods
+     *
+     * @return {Promise}
+     */
+    function loadAbsencePeriods() {
+      return AbsencePeriod.all()
+        .then(function (absencePeriods) {
+          vm.absencePeriods = absencePeriods;
+          vm.filters.leaveRequest.selectedPeriod = _.find(vm.absencePeriods, function (period) {
+            return period.current === true;
+          });
+        });
+    }
+
+    /**
+     * Loads the absence types
+     *
+     * @return {Promise}
+     */
+    function loadAbsenceTypes() {
+      return AbsenceType.all()
+        .then(function (absenceTypes) {
+          vm.absenceTypes = absenceTypes;
+        });
+    }
+
+    /**
+     * Loads all requests
+     *
+     * @param {int} page - page number
+     * @return {Promise}
+     */
+    function loadAllRequests(page) {
+      vm.pagination.page = page;
+      vm.loading.content = true;
+      Contact.all(contactFilters(), {
+        page: 1,
```

Comment:
This is to load all the records, it is the the only way I found to load everything at once